src/kudu/consensus/consensus_queue.cc

```diff
@@ -747,6 +755,8 @@ Status PeerMessageQueue::RequestForPeer(const string& uuid,
         *next_hop_uuid = uuid;
       }
     }
+
+    *enforce_response_rpc_token = enforce_response_rpc_token_;
```

Comment:
why is this required. We seem to be just passing a pointer and getting back the local variable.

Reply:
Yes, because one is a queue state and one is a request state. We copy the queue state for this particular request, which is the job of this function. We create a request for Peer using the current queue state.

The queue can move on after, but the request params are snapshotted.